src/classes/CON_Email.cls

```diff
@@ -43,103 +43,184 @@ public class CON_Email {
     */
 	public static void processPreferredEmail(Contact contact, List<Contact> oldlist) {
 
+        Boolean validatePreferred = !UTIL_CustomSettingsFacade.getSettings().Disable_Preferred_Email_Enforcement__c;
+
         // No need to run if we are in a delete context
         if(contact == null) {
             return;
         }
 
 		Boolean isUpdateOrDelete = (oldlist != null) ? true : false;
 
-	    // Store default email field mappings to preferred email pick list.
-	    Map<String,String> preferredEmailMappings = new Map<String,String>{   
-	        'University' => 'UniversityEmail__c',
-	        'Work' => 'WorkEmail__c',
-	        'Alternate' => 'AlternateEmail__c'
-	    };
+        // Build the list of email fields
+        CON_EmailFieldList emailFields = new CON_EmailFieldList(contact);
 
-	    Map<String,String> valuedEmailMap = getEmailFieldsWithValue(contact);
+        if( emailFields.valuedCount() > 0 ) {
 
-        if( valuedEmailMap.size() > 0 ) {
+            // Check the presence of a special cases when running a batch process.
+            // This will prevent some validation errors below to allow batches to complete successfully.
+            if( validatePreferred && System.isBatch() ) {
+                if( !emailFields.valueExists(contact.Email) && String.isBlank(contact.Preferred_Email__c)) {
```

Comment:
Great question, it is not necessary to check, because te next check overrides the preferred email, but to make it clearer what is happening I agree we should add the check for the blank email here.